sending/src/main/java/org/kin/sending/view/ContactDialogFragment.java

```diff
@@ -0,0 +1,220 @@
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.constraint.Group;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.Gravity;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.kin.sending.R;
+import org.kin.sending.presenter.ContactDialogPresenter;
+import org.kin.sending.presenter.ContactDialogPresenterImpl;
+import org.kin.sendkin.core.view.BaseWideDialogFragment;
+import org.kin.sendkin.core.view.SendKinEditText;
+import org.kin.sendkin.core.view.Utils;
+
+import java.util.UUID;
+
+public class ContactDialogFragment extends BaseWideDialogFragment implements ContactDialogView {
+    public static final String TAG = ContactDialogFragment.class.getSimpleName();
+
+    private static final String KEY_ID = "KEY_ID";
+    private ImageView deleteIcon;
+    private TextView title, addressNotValidWarning, saveBtn, deleteSubtitle;
+    private ContactDialogPresenter presenter;
+    private SendKinEditText nameInput, addressInput;
+    private Group nameAddressLayout;
+
+    private TextWatcher nameWatcher = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+        }
+
+        @Override
+        public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+        }
+
+        @Override
+        public void afterTextChanged(Editable editable) {
+            nameInput.clearError();
+            presenter.onNameChanged(editable.toString().trim());
+        }
+    };
+
+    private TextWatcher addressWatcher = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+        }
+
+        @Override
+        public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+        }
+
+        @Override
+        public void afterTextChanged(Editable editable) {
+            addressInput.clearError();
+            addressNotValidWarning.setVisibility(View.INVISIBLE);
+            presenter.onAddressChanged(editable.toString().trim());
+        }
+    };
+
+    public static ContactDialogFragment getInstance(@NonNull UUID id) {
+        ContactDialogFragment dialog = new ContactDialogFragment();
+        Bundle args = new Bundle();
+        //args.putInt(KEY_STEP, step);
+        args.putSerializable(KEY_ID, id);
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    public static ContactDialogFragment getInstance() {
+        return new ContactDialogFragment();
+    }
+
+    @Override
+    protected int getLayoutRes() {
+        return R.layout.contact_dialog_fragment;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        if (!(getActivity() instanceof SendKinView)) {
```

Comment:
I think you should do those kinds of checks in  onAttach() method or other method